### ISSUE TITLE
Build fails on Ubuntu 14.04 64-bit (gcc 4.8.4)

### ISSUE THREAD

**Issue by thedevilfish** [aws/aws-sdk-cpp#194]
Hello,

Doing a build fails on Ubuntu 14.04 for v0.12.3 of the SDK:

Ran the following commands:
```
cmake -DBUILD_ONLY="s3"
make
```

Here is the output from the build:
```
[ 28%] Building CXX object aws-cpp-sdk-core-tests/CMakeFiles/runCoreUnitTests.dir/utils/crypto/SymmetricCiphersTest.cpp.o
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp: In function ‘void TestGCMMultipleBuffers(const String&, const String&, const String&, const String&, const String&)’:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:498:70: error: call of overloaded ‘CryptoBuffer(<brace-enclosed initializer list>)’ is ambiguous
     auto decryptResult = CryptoBuffer({ std::move(decryptedStreams) });
                                                                      ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:498:70: note: candidates are:
In file included from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/crypto/Factories.h:19:0,
                 from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:17:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:246:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Utils::CryptoBuffer&&)
             CryptoBuffer(CryptoBuffer&& other) : ByteBuffer(std::move(other)) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:245:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const Aws::Utils::CryptoBuffer&)
             CryptoBuffer(const CryptoBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:244:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const ByteBuffer&)
             CryptoBuffer(const ByteBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:243:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Vector<Aws::Utils::Array<unsigned char>*>&&)
             CryptoBuffer(Aws::Vector<ByteBuffer*>&& toMerge) : ByteBuffer(std::move(toMerge)) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp: In function ‘void TestCBCMultipleBlockBuffers(const String&, const String&, const String&, const String&)’:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:567:70: error: call of overloaded ‘CryptoBuffer(<brace-enclosed initializer list>)’ is ambiguous
     auto decryptResult = CryptoBuffer({ std::move(decryptedStreams) });
                                                                      ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:567:70: note: candidates are:
In file included from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/crypto/Factories.h:19:0,
                 from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:17:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:246:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Utils::CryptoBuffer&&)
             CryptoBuffer(CryptoBuffer&& other) : ByteBuffer(std::move(other)) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:245:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const Aws::Utils::CryptoBuffer&)
             CryptoBuffer(const CryptoBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:244:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const ByteBuffer&)
             CryptoBuffer(const ByteBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:243:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Vector<Aws::Utils::Array<unsigned char>*>&&)
             CryptoBuffer(Aws::Vector<ByteBuffer*>&& toMerge) : ByteBuffer(std::move(toMerge)) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp: In function ‘void TestCTRMultipleBlockBuffers(const String&, const String&, const String&, const String&)’:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:629:70: error: call of overloaded ‘CryptoBuffer(<brace-enclosed initializer list>)’ is ambiguous
     auto decryptResult = CryptoBuffer({ std::move(decryptedStreams) });
                                                                      ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:629:70: note: candidates are:
In file included from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/crypto/Factories.h:19:0,
                 from /home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core-tests/utils/crypto/SymmetricCiphersTest.cpp:17:
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:246:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Utils::CryptoBuffer&&)
             CryptoBuffer(CryptoBuffer&& other) : ByteBuffer(std::move(other)) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:245:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const Aws::Utils::CryptoBuffer&)
             CryptoBuffer(const CryptoBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:244:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(const ByteBuffer&)
             CryptoBuffer(const ByteBuffer& other) : ByteBuffer(other) {}
             ^
/home/dev/Downloads/aws-sdk-cpp/aws-cpp-sdk-core/include/aws/core/utils/Array.h:243:13: note: Aws::Utils::CryptoBuffer::CryptoBuffer(Aws::Vector<Aws::Utils::Array<unsigned char>*>&&)
             CryptoBuffer(Aws::Vector<ByteBuffer*>&& toMerge) : ByteBuffer(std::move(toMerge)) {}
             ^
make[2]: *** [aws-cpp-sdk-core-tests/CMakeFiles/runCoreUnitTests.dir/utils/crypto/SymmetricCiphersTest.cpp.o] Error 1
make[1]: *** [aws-cpp-sdk-core-tests/CMakeFiles/runCoreUnitTests.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by rabinnh** [comment]
Having the identical issue.

**Comment by thedevilfish** [comment]
@JonathanHenson, any timeline when Fix gcc 4.8.5 build #196 will be merged/committed into master?
